### PR TITLE
Move UNALIGNED_OK detection to compile time instead of configure time.

### PR DESCRIPTION
**Pull request by nmoinvaz**
Pros: 
* It moves detection to one place instead of duplicating similar logic in both build scripts.

### CODE REVIEW

**Comment by nmoinvaz** [comment]
Rebased.

**Comment by nmoinvaz** [comment]
@Dead2 thoughts on this one if I were to make it configurable again? Like have some kind of `NO_UNALIGNED` that is set in the build scripts? And is checked in the header before the `UNALIGNED_OK` determination..

**Comment by nmoinvaz** [comment]
Rebased and added back configuration.

File: .github/workflows/cmake.yml [hunk 1]
```diff
@@ -41,11 +41,11 @@ jobs:
             cmake-args: -DZLIB_COMPAT=ON -DZLIB_SYMBOL_PREFIX=zTest_
             codecov: ubuntu_gcc_compat_sprefix
 
-          - name: Ubuntu GCC OSB -O1 No Unaligned64 UBSAN
+          - name: Ubuntu GCC OSB -O1 UBSAN
             os: ubuntu-latest
             compiler: gcc
             cxx-compiler: g++
-            cmake-args: -DWITH_UNALIGNED=ON -DUNALIGNED64_OK=OFF -DWITH_SANITIZER=Undefined
+            cmake-args: -DWITH_SANITIZER=Undefined
```
Comment (Dead2) [inline .github/workflows/cmake.yml:48]: We should still run tests for this mixed case, as that is what an i386-i686 would run, so I guess that would mean adding a new test that compiles and runs i686 for example.

Comment (nmoinvaz) [reply .github/workflows/cmake.yml:48]: I thought we had such a case with mingw

Comment (nmoinvaz) [reply .github/workflows/cmake.yml:48]: I have added a new test for `-m32`, hopefully that should be sufficent.

**Review by Dead2 — APPROVED** [review]
LGTM, needs rebase though

**Comment by nmoinvaz** [comment]
Rebased.

**Comment by nmoinvaz** [comment]
Rebased again.

**Pull request merged**